snippets/single-product.liquid

```diff
@@ -116,39 +116,42 @@
 </div>
 
 <script defer>
-(function() {
-  const products = {{ product.images.size }};
-  const desktopSlider = new Splide(`#yc_slider_{{ uniqID }}.yc-slider__desktop`, {
-    arrows: products > 3,
+(function () {
+const products = {{ product.images.size }};
+const ycSlider = $('#yc_slider_{{ uniqID }}');
+
+if (ycSlider) {
+  const mobileSlider2 = new Splide(`#yc_slider_{{ uniqID }}`, {
+    arrows: false,
+    drag: true,
+    flickMaxPages: 1,
+    pagination: true,
+    paginationType: 'bullets',
+    gap:'4px',
+    perPage: 1,
     rewind: true,
-    perPage: Math.min(products, 4),
-    perMove: 1,
-    gap: '10px',
-    drag: 'free',
-    snap: true,
-    breakpoints: {
-      1024: {
-        perPage: Math.min(products, 3),
-      },
-      767: {
-        direction: 'ttb',
-        height: '255px',
-      },
-    },
+    fixedWidth: '90%',
+    type: 'loop',
+    rewind: true,
+    rewindSpeed: 1000
   });
-  const resizedIcons = () => {
-    if (products >= 3 && window.innerWidth < 1024 || products >= 4) {
-      document.querySelector('.yc-slider__desktop').style.padding = '0 40px';
-    }
-    if (products > 3) {
-      document.querySelector('.thumbnail-images').style.justifyContent = 'center';
-    }
-    if (window.innerWidth < 768) {
-      document.querySelector('.yc-slider__desktop').style.padding = '0';
+
+  mobileSlider2.mount();
```

Comment:
It's better to wrap this logic inside a function to create a new execution context and avoid naming variables with numbers

Reply:
it is already inside an anonymous function

Reply:
name updated